source/slang/check.cpp

```diff
@@ -1833,24 +1837,22 @@ namespace Slang
 
             // Make sure that by-name lookup is possible.
             buildMemberDictionary(typeDeclRef.getDecl());
-
-            Decl* firstMemberOfName = nullptr;
-            typeDeclRef.getDecl()->memberDictionary.TryGetValue(name, firstMemberOfName);
-
-            if (!firstMemberOfName)
+            auto lookupResult = lookUpLocal(getSession(), this, name, typeDeclRef);
+            
+            if (!lookupResult.isValid())
             {
                 getSink()->diagnose(inheritanceDecl, Diagnostics::typeDoesntImplementInterfaceRequirement, typeDeclRef, requiredMemberDeclRef);
                 return nullptr;
             }
 
             // Iterate over the members and look for one that matches
             // the expected signature for the requirement.
-            for (auto memberDecl = firstMemberOfName; memberDecl; memberDecl = memberDecl->nextInContainerWithSameName)
+            for (auto member : lookupResult)
             {
-                if (doesMemberSatisfyRequirement(DeclRef<Decl>(memberDecl, typeDeclRef.substitutions), requiredMemberDeclRef, requirementWitness))
-                    return memberDecl;
+                if (doesMemberSatisfyRequirement(member.declRef, requiredMemberDeclRef, requirementWitness))
+                    return member.declRef.getDecl();
```

Comment:
The other gotchas here are that we are throwing away all the information on how we found the declaration. This includes any generic specialization that might have been applied (e.g., to specialize a generic extension so that it matches the given type), and then also any “breadcrumbs” that represent the result being found by a non-trivial lookup step (including via an inheritance decl).

In the case where we end up satisfying a requirement via a default implementation inherited from an interface, we’d expect the decl ref (and/or the breadcrumbs) to contain the info we need to specialize the default implementation for the chosen “this type.”